### PR TITLE
11 - allUpperCase

### LINKED ISSUE #11: allUpperCase.sh
Listar lexicográficamente las palabras que se presenten con todas sus letras en mayúsculas (no repetir palabras en el listado).
Ejemplos: EJEMPLO, CANCIÓN

### PR DESCRIPTION
**Pull request by fedeantuna**

### CODE REVIEW

**Review by salvanya — COMMENTED** [review]
It doesn't consider all upercase words in the same line. It only shows the first one

File: allUpperCase.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
```
Comment (salvanya) [inline allUpperCase.sh]:
```suggestion
#!/bin/bash
```
This is not the shebang we agreed to use. For consistency I recommend to change it to this one.

File: allUpperCase.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+TEXT=$(cat $1)
+
+for WORD in $TEXT
+do
+    if [[ $WORD =~ [[:upper:]]+$ ]]; then
+        echo $WORD
+    fi
```
Comment (Flor243) [inline allUpperCase.sh]:
```suggestion
   [[ $WORD =~ [0-9] ]] && continue
    SANITIZED_WORD=$(echo $WORD | tr -d '.')
    [[ $SANITIZED_WORD =~ [[:upper:]]+$ ]] && echo $SANITIZED_WORD
```
With this we can ignore "words" that contain numbers and then make sure that we remove unnecessary dots 🙂

File: allUpperCase.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
```
Comment (salvanya) [inline allUpperCase.sh:6]: This is where documentation should be :'(

Comment (fedeantuna) [reply allUpperCase.sh:6]: done :)

**Review by Flor243 — APPROVED** [review]

**Review by salvanya — APPROVED** [review]

**Pull request merged**